backend/GameManager.py

```diff
@@ -58,5 +82,173 @@ def get_pieces_left(self):
         }
 
     def end_turn(self):
-        self.turn = Turn.swap_turn(self.turn)
+        self.turn = Color.swap_turn(self.turn)
         return 1
+
+    def is_adjacent(self, current_row, current_col, target_row, target_col):
```

Comment:
Need to add the function I linked to here.

Reply:
not sure what you want on this